### DISCUSSION TITLE
trying to stream djing

### DISCUSSION THREAD

**Discussion by JameHW123** [ExistentialAudio/BlackHole#104] [General]
currently want to stream through OBS from Traktor Pro 3 software, I need a way to send audio internally to obs, as well as to 2 studio monitors and headphones. I have managed to send to monitors and obs but not to the headphones, so almost appears as if I need availability to another output or something. 

anyone got any ideas?

**Comment by krono-i2** [comment]
Are you on macOS?

**Comment by JameHW123** [comment]
Yes Mac Catalina

**Comment by krono-i2** [comment]
Are you using a 4-ch audio interface?
I suppose 2 channels for main output and 2 channels for the headphones.

**Comment by JameHW123** [comment]
Currently not using and audio interface, I am going through my decks for doing traktor s4 mk3 

Here are all the stats taken from online about outputs on the decks, currently looking at using line outputs and headphone output 

AUDIO OUTPUTS (D/A)
Channels	2 Stereo
Sample Rate / Resolution	96-kHz / 24-bit
Converter	Cirrus Logic CS4385
LINE OUTPUTS (XLR & 1/4" BOOTH)
DNR (a-weighted)	110 dB
Frequency Response	20 Hz - 20 kHz, +0.4/- 0.2 dB
THD+N	0.001%
LINE OUTPUTS (RCA)
DNR (a-weighted)	110 dB
Frequency Response	20 Hz - 20 kHz, +0.6/- 0.2dB
THD+N	0.001%
HEADPHONE OUTPUT
Load Impedance	33Ω
DNR (a-weighted)	104 dB
Frequency Response	20 Hz - 20 kHz, +0.4/- 0.3 dB

**Comment by krono-i2** [comment]
Ok, kontrol s4 has an internal audio interface.
You can solve in some pretty simple steps:
1) Create an aggregate device in Audio/MIDI configuration, and select kontrol s4 (first, master clock) and blackhole (second).
2) In traktor select the new aggregate device as audio interface, in output routing select “internal” mode and set channels 1-2 for master, 3-4 for monitor, and 5-6 for rec output (it should correspond to blackhole).
3) In OBS select blackhole as live audio input.

**Comment by JameHW123** [comment]
Where is the input routing button?

**Comment by krono-i2** [comment]
> Where is the input routing button?
> […](#)

“Output routing” tab is after “Audio setup” in traktor preferences.
